test/model/message_test.dart

```diff
@@ -269,6 +271,48 @@ void main() {
     test('rendering-only update does not change timestamp (for old server versions)', () async {
       await checkRenderingOnly(legacy: true);
     });
+
+    group('Handle message edit state update', () {
+      final message = eg.streamMessage();
+
+      Future<void> sendEvent(Message message, UpdateMessageEvent event) async {
+        await prepare();
+        await prepareMessages([message]);
+
+        await store.handleEvent(event);
+        checkNotifiedOnce();
+      }
+
+      test('message not moved update', () async {
+        await sendEvent(message, eg.updateMessageMoveEvent(message));
+        check(store).messages.values.single.editState.equals(MessageEditState.none);
+      });
+
+      test('message topic moved update', () async {
+        await sendEvent(message, eg.updateMessageMoveEvent(message,
+          origTopic: 'old topic',
+          newTopic:  'new topic',
+        ));
+        check(store).messages.values.single.editState.equals(MessageEditState.moved);
+      });
+
+      test('message topic resolved update', () async {
+        await sendEvent(message, eg.updateMessageMoveEvent(message,
+          origTopic: 'new topic',
+          newTopic:  '✔ new topic',
+        ));
+        check(store).messages.values.single.editState.equals(MessageEditState.none);
+      });
+
+      test('message stream moved update', () async {
+        await sendEvent(message, eg.updateMessageMoveEvent(message,
+          origTopic: 'topic',
+          newTopic: 'topic',
+          newStreamId:  20,
+        ));
+        check(store).messages.values.single.editState.equals(MessageEditState.moved);
+      });
```

Comment:
Hmm, this test is failing in the current revision. I think it's because the event-handling code unconditionally sets the message's state to `MessageEditState.edited`, in `_handleUpdateMessageEventContent`—that doesn't seem desirable 🙂—and then `_handleUpdateMessageEventMove` leaves that in place; here's from `_handleUpdateMessageEventMove`:

```dart
      // Do not override the edited marker if the message has also been moved.
      if (message.editState == MessageEditState.edited) continue;
```

Reply:
Modified `_handleUpdateMessageEventContent` to check `origContent` so that we know if there has been an edit. This should make the message of id `messageId` "edited", but the remaining messages of ids from `messageIds` remain as "moved".